parsl/data_provider/staging.py

```diff
@@ -0,0 +1,68 @@
+from concurrent.futures import Future
+from typing import Optional, Callable
+from parsl.app.futures import DataFuture
+from parsl.data_provider.files import File
+
+from typing import TYPE_CHECKING
+if TYPE_CHECKING:
+    from parsl.data_provider.data_manager import DataManager
+
+
+class Staging:
```

Comment:
It seems we mix `Staging` and `Scheme` (i.e. `GlobusScheme` but `FTPSeparateTaskStaging`, and use 'staging provider/staging scheme' interchangeably) throughout. Do they mean different things? If not, can we pick one and remove the other?

Reply:
The word "scheme" in file staging has had two meanings: one for the URL scheme (the bit before the colon, which is implicitly `file:` if not specified), and the other for the code that started off as GlobusScheme and has subsequently evolved into what I'm calling staging providers. I've added a commit that tries to distinguish the two uses of the term, keeping "scheme" for URLs, but using "provider" for the second case.